java/ql/src/Advisory/Java Objects/AvoidCloneMethodAccess.ql

```diff
@@ -11,14 +11,10 @@
 
 import java
 
-class ObjectCloneMethod extends Method {
-  ObjectCloneMethod() {
-    this.getDeclaringType() instanceof TypeObject and
-    this.getName() = "clone" and
-    this.hasNoParameters()
-  }
-}
-
-from MethodAccess ma, ObjectCloneMethod clone
-where ma.getMethod().overrides(clone)
+from MethodAccess ma, Method m
+where
+  m = ma.getMethod() and
+  m instanceof CloneMethod and
```

Comment:
This might a behavior change because previously it would only consider calls to methods directly overriding `Object.clone`, while now it considers transitive overrides as well.
However the way the query description is written suggests that this is desired.